src/messages/TextWithButtons/TextWithButtons.tsx

```diff
@@ -31,19 +32,31 @@ const TextWithButtons: FC = () => {
 	const shouldBeDisabled = isQuickReplies && messageParams?.hasReply;
 	const modifiedAction = shouldBeDisabled ? undefined : action;
 
+	const webchatButtonTemplateTextId = getRandomId("webchatButtonTemplateHeader");
```

Comment:
On the other components I implemented this function, I wrapped it in useMemo hook, if required. In this case I just missed, and it's the only case.
It's also good idea to create a hook for it and reuse everywhere. I will add it.

Reply:
There is same usage in DatePicker.tsx. 

If you implement it as a helper hooks, let's avoid useMemo for having unnecessary overhead. 

useState or useRef should be sufficient here.

Reply:
I disagree with you when you are discouraging useMemo usage, in this specific case. You have already seen I used it with empty array as deps, this is the key of the discussion here. It would be enough to have a look on the source code ([ReactFiberHooks.js](https://github.com/facebook/react/blob/540bab085d571789f4562565eebfd0db9f36345c/packages/react-reconciler/src/ReactFiberHooks.js)) to notice that mountState and mountMemo are doing the same job. They just invoke a instance of "Hook" and store the value returned by the function (in our case getRandomId) on the memoizedState field. That's it. On all updates and rerenders, they do nothing else then return that value, without any additional calculation. It means these two line of code are equivalent in term of complexity and performance:

```
const [randomId] = useState(getRandomId);
const randomId = useMemo(getRandomId, []);
```

A completely different scenario would be in case of dependecies on useMemo array. In that case need to be careful because of risks of compromising the app performance (see areHookInputsEqual function). That's why need to use it with deps only if very necessary and following the best practices. The same we can say about all the other hooks that have dependencies.

Furthermore, in our specific use case that is accessibility attributes, it was created a specific hook called [useId()](https://react.dev/reference/react/useId). So if really we want to be picky, none of the hooks mentioned above are the best ones for our use case. Unfortunately we cannot implement useId() hook because it was introduced in React 18 and our consumer Apps run an older version.

Said that, on my last commit I integrated the custom hook useRandomId() exactly as we have have done in WebchatWidget, it's enough for our case. We used useState there, but using useMemo with empty array deps would be an equivalent solution.
Hope this explanation helps for the future.